src/main/java/com/mentoring/ecommerce/adapter/in/web/ProductController.java

```diff
@@ -30,35 +36,47 @@ public class ProductController {
 
     private final ProductMapper productMapper;
 
-    @PostMapping()
-    @ResponseStatus(HttpStatus.CREATED)
-    public void saveProduct(@Valid @RequestBody final ProductRequest request) {
-        saveUseCase.saveProduct(productMapper.toDomain(request));
+    @PostMapping
+    @ResponseStatus(CREATED)
+    public ProductResponse saveProduct(@RequestBody final ProductRequest request) {
```

Comment:
I noticed that in your project you use the @Valid annotation, I was testing it but I still need to research to understand better.